docs/design/daemon.md

```diff
@@ -0,0 +1,128 @@
+# Daemons in Kubernetes
+
+**Author**: Ananya Kumar (@AnanyaKumar)
+
+**Status**: Draft proposal; prototype in progress.
+
+This document presents the design of a daemon controller for Kubernetes, outlines relevant Kubernetes concepts, describes use cases, and lays out milestones for its development.
+
+## Motivation
+
+In Kubernetes, a Replication Controller ensures that the specified number of a specified pod are running in the cluster at all times (pods are restarted if they are killed). With the Replication Controller, users cannot control which nodes their pods run on - Kubernetes decides how to schedule the pods onto nodes. However, many users want control over how certain pods are scheduled. In particular, many users have requested for a way to run a daemon on every node in the cluster, or on a certain set of nodes in the cluster. This is essential for use cases such as building a sharded datastore, or running a logger on every node. In comes the daemon controller, a way to conveniently create and manage daemon-like workloads in Kubernetes.  
```

Comment:
It's not really true that "users cannot control which nodes their pods run on" -- you can use node selector in your Pod.Spec. The reason why ReplicationController is sub-optimal for this use case is that you can't know what value to set for `replicas` in the RC spec, since the size of the cluster (or size of the subset of the cluster defined by the node selector) may vary over time.